pandas/types/common.py

```diff
@@ -491,3 +491,27 @@ def pandas_dtype(dtype):
         return dtype
 
     return np.dtype(dtype)
+
+
+def _is_fillable_value(value):
+    pandas_ts_types = ('Timestamp', 'Period', 'Timedelta')
+    pandas_block_types = ('Series', 'DataFrame')
+
+    if any([isinstance(value, (list, dict)),
```

Comment:
Missed these `is_list_like`/`is_dict_like` helpers, these are indeed important. But, why the evaluation that comes afterwards? As I understand it, we should be rejecting list and dict type inputs outright. 

The former is valid in `fillna`, though passing a dict isn't implemented in any of the `fill_value` parameters. Lists, meanwhile, are never a valid fill value.

What I mean is that I think the implementation would be something like:

```
def validate_fill_value(value):
   def _validate(v):
        # do validation on a scalar
        return boolean
   if is_list_like(value) or is_dict_like(value):
       return False
   else:
       return _validate(value)
```